packages/webapp/src/pages/index.tsx

```diff
@@ -155,58 +128,106 @@ const ElectionCard = () => {
+            <Link
+                href="http://eden.eoscommunity.org"
+                target="_blank"
+                isExternal
+            >
+                eden.eoscommunity.org
+            </Link>
+            .
+        </Text>
+    </>
+);
 
 interface CommunityStatsProps {
-    memberStats: any;
-    treasuryBalance: Asset | undefined;
+    memberStats: UseQueryResult<MemberStats | undefined>;
+    treasuryStats: UseQueryResult<Asset | undefined>;
 }
 
-const CommunityStatsCard = ({
-    memberStats,
-    treasuryBalance,
-}: CommunityStatsProps) =>
-    memberStats && treasuryBalance ? (
-        <Card className="flex flex-col justify-center items-center h-full space-y-4 lg:text-lg">
+const CommunityStats = ({
+    memberStats: memberInfo,
+    treasuryStats,
+}: CommunityStatsProps) => {
+    const { data: memberStats, isLoading: isLoadingMemberStats } = memberInfo;
+    const {
+        data: treasuryBalance,
+        isLoading: isLoadingTreasuryBalance,
+    } = treasuryStats;
+
+    if (isLoadingMemberStats || isLoadingTreasuryBalance) {
+        return <LoadingCard />;
+    }
+
+    if (!memberStats && !treasuryBalance) return null;
+
+    return (
+        <Card className="flex flex-col justify-center items-center space-y-1 lg:text-lg my-4">
             <Heading size={2} className="mb-2">
                 Community Stats
             </Heading>
-            <Text className="font-medium" size="inherit">
-                Treasury: {assetToString(treasuryBalance, 4)}
-            </Text>
-            <Link href={ROUTES.MEMBERS.href} className="font-medium">
-                {memberStats.active_members} active member
-                {memberStats.active_members !== 1 && "s"}
-            </Link>
-            <Link
-                href={`${ROUTES.INDUCTION.href}/pending-invitations`}
-                className="font-medium"
-            >
-                {memberStats.pending_members} pending invitation
-                {memberStats.pending_members !== 1 && "s"}
-            </Link>
+            {treasuryBalance && (
+                <Text className="font-medium" size="inherit">
+                    Treasury: {assetToString(treasuryBalance, 4)}
+                </Text>
+            )}
+            {memberStats && (
+                <>
+                    <Link href={ROUTES.MEMBERS.href} className="font-medium">
+                        {memberStats.active_members} active member
+                        {memberStats.active_members !== 1 && "s"}
+                    </Link>
+                    <Link
+                        href={`${ROUTES.INDUCTION.href}/pending-invitations`}
+                        className="font-medium"
+                    >
+                        {memberStats.pending_members} pending invitation
+                        {memberStats.pending_members !== 1 && "s"}
+                    </Link>
+                </>
+            )}
         </Card>
-    ) : null;
+    );
+};
```

Comment:
thought: I just realized how big this page is. I think we should keep following the pattern of having nextjs `pages` lean, only responsible for routing and prefetching and have the components in their corresponding domain package. (as we are doing with elections, inductions, treasury etc.)

Reply:
Yes, it is too big. I'll break it apart as part of my Home page work I'm finishing today. And thank you for the review!!!

Reply:
Will tackle shortly.